File: Array/xietao/6月4号—数组融合/java/MeSolution.java
public class MeSolution {
        public void merge(int[] A, int m, int[] B, int n) {
            for(int i = 0; i<n;i++){
                int b = B[i];
                int j = m - 1 + i;
                while(j >= 0 && b <= A[j]){
                    A[j+1] = A[j];
                    j--;
                }
                A[j+1] = b;
            }
        }
    }

File: Array/xietao/6月16号有效的括号/Kotlin/MeSolution.kt
package xt

import java.util.*

class MeSolution {
    fun isValid(s: String): Boolean {
        if (s.length % 2 == 1) {
            return false
        }
        val stack: Stack<Char> = Stack<Char>()
        for (i in 0 until s.length) {
            val c = s[i]
            if (c == '{' || c == '[' || c == '(') {
                stack.push(c)
            } else {
                if (stack.empty()) {
                    return false
                }
                val preChar = stack.peek()
                if (preChar == '{' && c == '}' ||
                        preChar == '[' && c == ']' ||
                        preChar == '(' && c == ')') {
                    stack.pop()
                } else {
                    return false
                }
            }
        }
        return stack.empty()
    }
}